lambda.tf

```
@@ -0,0 +1,68 @@
+# ------------------------------------------------------------------------------
+# The AWS Lambda function that is used to publish egress IP addresses.
+# The ZIP file is created with:
+# http://github.com/cisagov/publish-egress-ip-lambda
+# ------------------------------------------------------------------------------
+
+resource "aws_lambda_function" "publish_egress_ip" {
+  provider = aws.deploy
+
+  description      = var.lambda_function_description
+  filename         = var.lambda_zip_filename
+  function_name    = var.lambda_function_name
+  handler          = "lambda_handler.handler"
+  memory_size      = 128
+  role             = aws_iam_role.lambdaexecution_role.arn
+  runtime          = "python3.9"
+  source_code_hash = filebase64sha256(var.lambda_zip_filename)
+  timeout          = 600
+}
+
+# The CloudWatch log group for the Lambda function
+resource "aws_cloudwatch_log_group" "lambda_logs" {
+  provider = aws.deploy
+
+  name              = format("/aws/lambda/%s", var.lambda_function_name)
+  retention_in_days = 30
+}
+
+# Schedule the Lambda function to run every X minutes
+resource "aws_cloudwatch_event_rule" "lambda_schedule" {
+  provider = aws.deploy
+
+  description         = format("Executes %s Lambda every %d minutes", var.lambda_function_name, var.lambda_schedule_interval)
+  name                = format("%s-every-%d-minutes", var.lambda_function_name, var.lambda_schedule_interval)
+  schedule_expression = format("rate(%d minutes)", var.lambda_schedule_interval)
```

Per the documentation:
> If the value is equal to 1, then the unit must be singular. Similarly, for values greater than 1, the unit must be plural. For example, rate(1 hours) and rate(5 hour) are not valid, but rate(1 hour) and rate(5 hours) are valid.

If you have another way you'd prefer to do this feel free.

```suggestion
  schedule_expression = format("rate(%d minute%s)", var.lambda_schedule_interval, var.lambda_schedule_interval != 1 ? "s" : "")
```

---

That works for me- nice catch!  After seeing your suggestion, I briefly thought about restricting `var.lambda_schedule_interval` to be greater than 1, but that's pretty lame. 😄